### ISSUE TITLE
[HOLD for payment 2024-03-22] [$500] [MEDIUM] Approved expense preview does not show GBR when submitter needs to add a bank account 

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#34278]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @anmurali 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1704819780539799

## Action Performed:
Create a money request and submit it to a Collect workspace. Approve and pay as the workspace admin.

## Expected Result: 
After an expense is approved, the submitting user should see a GBR in the expense preview which prompts them to add a bank account by clicking in to reveal the `Add bank account` CTA. 

## Actual Result: 
After an expense is approved, the submitting does not see a GBR in the expense preview, and there is no indicator that they must click in to find the `Add bank account` CTA. 

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![File (2)](https://github.com/Expensify/App/assets/38435837/8ca65b45-073f-48f3-90ea-466cc0545d3a)

![File (3)](https://github.com/Expensify/App/assets/38435837/850e808d-f86a-48ac-9b02-fd34fb171c0d)

![IMG_0239](https://github.com/Expensify/App/assets/38435837/de3deee5-9dfd-4668-9e1f-7f0474c19f6d)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e7910dca0f3c5b57</li>
        <li>Upwork Job ID: 1747385136600576000</li>
        <li>Last Price Increase: 2024-01-23</li>
<li>Automatic offers: </li>
<ul>
<li>cubuspl42 | Reviewer | 28131009</li>
<li>tienifr | Contributor | 28131010</li>
</ul></li>
    </ul>
</details>

**Comment by stephanieelliott** [comment]
Hey @m-natarajan were you able to reproduce this, or should we add the `Needs Reproduction` label?

**Comment by happy-man** [comment]
Hello. I want to work on the issue but I'm unable to do  initial `npm install`.
I've got such error
![image](https://github.com/Expensify/App/assets/7570979/da01da84-5591-4e90-afea-200210e6cd8e)
As I understand from error messages project's version of react is 18 and @storybook package wants a version not higher than 17.
Please, help me resolve this issue properly. Thanks.

**Comment by cubuspl42** [comment]
@happy-man 

Welcome to Expensify! Please ensure to read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) and check out a few other issues with other contributors' proposals to get a feeling of how the Expensify process works. If you have any general questions unrelated to this issue, ask them on [`#expensify-open-source`](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) Slack channel.

**Comment by happy-man** [comment]
Contributor details
Your Expensify account email: rosilivejster@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01ecd21cecad2f34a0

**Comment by happy-man** [comment]
> @happy-man
> 
> Welcome to Expensify! Please ensure to read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) and check out a few other issues with other contributors' proposals to get a feeling of how the Expensify process works. If you have any general questions unrelated to this issue, ask them on [`#expensify-open-source`](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) Slack channel.

Hello! Thanks. How can I be added to slack channel?

**Comment by stephanieelliott** [comment]
@happy-man please read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md), this instructions for getting added to Slack are included there. 

> To request an invite to Slack, email [contributors@expensify.com](https://www.expensify.com/_support/?supportEmail=contributors%40expensify.com&comment=better%20supportal%20issue%20nubmer%3A%20CONTRIBUTING.md&location=_support/) with the subject Slack Channel Invites. We'll send you an invite!

**Comment by cubuspl42** [comment]
@happy-man 

Hi again!

Please ensure to read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md). Have you read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md)? If you have some questions, you can also check the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md). Sometimes, I feel that people don't read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md), but then I realize that it's unlikely, because two bots and four people ask them to read the [contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md).

**Comment by cubuspl42** [comment]
@tienifr 

> This is new feature, we never had it implemented.

Does this chat log clarify anything?

![image](https://github.com/Expensify/App/assets/2590174/3b1e2723-0343-4f68-9559-df67e0b6583e)

**Comment by stephanieelliott** [comment]
> From https://github.com/Expensify/App/issues/34278#issuecomment-1898153759, it's mentioned that the "specifically the description" is supposed to be in the preview. But the OP in this GH issue doesn't mention about description, it's just about GBR and CTA

I think we want the submitter's expense preview (which is shown in **screenshot 2**) to reflect the same expense details and CTA that are shown in the approved expense preview (which is shown in screenshot 1). So for this example, that would be: 

> - Cash > Approved 
> 	- $35
> 	- Ride for BDO conference 

Then when clicked, it would show the green `Add bank account` button (as shown in **screenshot 2**)

@anmurali can you confirm the above? 

> https://github.com/Expensify/App/issues/34278#issuecomment-1898153759 also doesn't mention anything about GBR, CTA when confirming it's a regression, so a bug so I'm not sure whether it's included in this issue's scope. Are the GBR, CTA issue mentioned in the OP a regression? Or are they new features we want to implement here.

This is reported to be a regression, and the scope of the issue is to: 1) update the expense preview to show the relevant details,  and 2) show the green `Add bank account` button when the expense preview is clicked. Maybe I am missing something here, but do we need to drill into whether this is 100% regression in order to solve it?

**Comment by stephanieelliott** [comment]
Chatted with @anmurali and it seems like this issue somehow got conflated with [another issue](https://expensify.slack.com/archives/C049HHMV9SM/p1704388820763769). 

For this issue, we're basically just adding a GBR to the expense preview so that the submitter knows to click in to find the existing`Add bank Account` CTA.  (which already exists in the Detail view). The detailed view is in fact out of scope, as the CTA already exists. 

I've updated the issue body, I think it's much more clear now but LMK if we need to drill into it further! cc'ing @tienifr since I gave you some wildly conflicting info earlier 😅 (sorry bout that!)

**Comment by cubuspl42** [comment]
Makes sense, I agree that there's not much to add. Let's figure out the details in the PR.

I approve the proposal by @tienifr 

C+ reviewed 🎀 👀 🎀

**Comment by stephanieelliott** [comment]
PR is under review

**Comment by stephanieelliott** [comment]
Seems like this is held up on some clarifying questions asked in the PR, I bumped that in Slack in case it was missed within the PR: https://expensify.slack.com/archives/C01GTK53T8Q/p1707427263037779

**Comment by cubuspl42** [comment]
As per the PR status, the most important feedback was [this thread](https://github.com/Expensify/App/pull/35486/files#r1474079405). I bumped @tienifr.

**Comment by stephanieelliott** [comment]
PR is being actively reviewed.

**Comment by stephanieelliott** [comment]
PR is still under active review!

**Comment by cubuspl42** [comment]
@stephanieelliott We're having problems with reproducing the context of this issue. It's partially my fault, as I should've caught this earlier.

Slack thread: https://expensify.slack.com/archives/C01GTK53T8Q/p1708946657174689?thread_ts=1708772104.743519&cid=C01GTK53T8Q

**Comment by cubuspl42** [comment]
This is on me now, I'll do the checklist today 👍

**Comment by stephanieelliott** [comment]
Looks like we are working through testing on the PR

**Comment by cubuspl42** [comment]
- The PR that introduced the bug has been identified. Link to the PR:
    -  Essentially it's a new feature fixing a "UX bug"
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
    - N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
    - No need for additional discussion
- Determine if we should create a regression test for this bug.
    - No
- If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
    - N/A

**Comment by stephanieelliott** [comment]
All paid! Summarizing payment on this issue: 

 - Contributor: @tienifr $500 PAID via Upwork 
 - Contributor+: @cubuspl42 $500 PAID via Upwork 
 
 Upwork job is here: https://www.upwork.com/jobs/~01e7910dca0f3c5b57
